x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotEnableAllocationDecider.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.searchablesnapshots;
+
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.routing.RecoverySource;
+import org.elasticsearch.cluster.routing.RoutingNode;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.cluster.routing.allocation.RoutingAllocation;
+import org.elasticsearch.cluster.routing.allocation.decider.AllocationDecider;
+import org.elasticsearch.cluster.routing.allocation.decider.Decision;
+import org.elasticsearch.cluster.routing.allocation.decider.EnableAllocationDecider;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+
+public class SearchableSnapshotEnableAllocationDecider extends AllocationDecider {
+
+    static final String NAME = "searchable_snapshots_enable";
+
+    /**
+     * This setting indicates the behavior of searchable snapshots when cluster.routing.allocation.enable=primaries
+     *
+     */
+    public static final Setting<EnableAllocationDecider.Allocation> SEARCHABLE_SNAPSHOTS_ALLOCATION_ENABLE_PRIMARIES_SETTING =
+        new Setting<>("xpack.searchable.snapshot.allocation.enable.primaries", EnableAllocationDecider.Allocation.NONE.toString(),
```

Comment:
Hmm this means we have these options:

```
xpack.searchable.snapshot.allocation.enable.primaries: none
xpack.searchable.snapshot.allocation.enable.primaries: primaries
```

The repetition of `primaries` seems odd to me. I see that this is a bit nicer than a straight boolean, given the relationship to the other allocation decider, but how about these options instead?

```
xpack.searchable.snapshot.allocation.enable: none
xpack.searchable.snapshot.allocation.enable: all
```

Reply:
I misunderstood the precise meaning of this setting, sorry, it defaults to `none`!

We discussed this in another channel and decided a boolean called `xpack.searchable.snapshot.allocate_on_rolling_restart` which defaults to `false`.